jvm/src/test/scala/ReadTreeSuite.scala

```diff
@@ -0,0 +1,67 @@
+import tastyquery.ast.Names.{SimpleName, QualifiedName, TypeName}
+import tastyquery.ast.Trees._
+import tastyquery.ast.Types.{TermRef, TypeRef}
+import tastyquery.reader.TastyUnpickler
+
+import java.nio.file.{Files, Paths}
+
+class ReadTreeSuite extends munit.FunSuite {
+  val ResourceProperty = "test-resources"
+
+  def unpickle(tastyBytes: Array[Byte]): List[Tree] = {
+    val unpickler = new TastyUnpickler(tastyBytes)
+    unpickler.unpickle(new TastyUnpickler.TreeSectionUnpickler()).get.unpickle
+  }
+
+  def getResourcePath(name: String): String =
+    s"${System.getProperty(ResourceProperty)}/$name.tasty"
+
+  def checkUnpickle(name: String, p: PartialFunction[Tree, Unit]): Unit =
+    test(name) {
```

Comment:
Consider writing those `test("strings") {}` at each call site instead. It's easier to read as a test suite.